core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -467,7 +474,7 @@ private[spark] class TaskSchedulerImpl private[scheduler](
       taskState: TaskState,
       reason: TaskFailedReason): Unit = synchronized {
     taskSetManager.handleFailedTask(tid, taskState, reason)
-    if (!taskSetManager.isZombie && taskState != TaskState.KILLED) {
+    if (!taskSetManager.isZombie) {
```

Comment:
@kayousterhout Are we making the change that killed tasks can/should be retried ?
If yes, this is a behavior change; and TSM.handleFailedTask(), we need to do the same.

This is what I mentioned w.r.t killed not resulting in task resubmission.